### PR TITLE
Improved Delay component

### PR DESCRIPTION
**Pull request by heniotierra**
At its current state, the "Delay" component do not do anything when the delay ends. This modification fixes that, allowing you to pass a "callbackOff" method that will be called after all iterations of delay are executed.

Also, now it's possible to cancel a delay with the method .cancelDelay, passing as argument a method reference, the same method reference passed to .delay.

Finally, fixed a minor inconsistency in the the comments.

### CODE REVIEW

**Comment by heniotierra** [comment]
Now I think it's good. :)

File: src/time.js [hunk 1]
```diff
@@ -45,12 +47,15 @@ Crafty.c("Delay", {
     /**@
      * #.delay
      * @comp Delay
-     * @sign public this.delay(Function callback, Number delay)
-     * @param callback - Method to execute after given amount of milliseconds
-     * @param delay - Amount of milliseconds to execute the method
+     * @sign public this.delay(Function callback, Number delay, Number repeat, Function callbackOff)
```
Comment (mucaho) [inline src/time.js]: I would emphasize the optionality of the last parameter (similar to rest of Crafty code)

```
@sign public this.delay(Function callback, Number delay, Number repeat[, Function callbackOff])
```

File: src/time.js [hunk 2]
```diff
@@ -68,15 +73,39 @@ Crafty.c("Delay", {
...
+     * #.cancelDelay
+     * @comp Delay
+     * @sign public this.cancelDelay(Function callback)
+     * @param callback - Method reference passed to .delay
+     *
+     * The cancelDelay method will cancel a delay set previously.
+     *
+     * @example
+     * ~~~
+     * this.cancelDelay( reference );
```
Comment (mucaho) [inline src/time.js]: could you add a short example here with an actual reference?
e.g.

``` javascript
var log = function() {
    console.log("100ms later");
};

console.log("start");
var ent = Crafty.e("Delay").delay(log, 100, 10);

... // sometime later in the future
ent.cancelDelay(log);
```

**Comment by mucaho** [comment]
Thanks for this PR. I added some minor suggestions, but other than that this looks good.

Would you also [squash your commits into one](https://github.com/craftyjs/Crafty/wiki/Workflow#squashinq-commits), please? This helps keep a clean history of the codebase.

**Comment by heniotierra** [comment]
@mucaho Done! Phew! And thanks for the suggestions! :))

**Comment by mucaho** [comment]
Great, merging.

**Pull request merged**